Applications/Spire/Source/TimeAndSales/TimeAndSalesTableView.cpp

```diff
@@ -0,0 +1,175 @@
+    style.get(Any()).
+      set(border_size(0)).
+      set(horizontal_padding(scale_width(2))).
+      set(vertical_padding(scale_height(1.5)));
+  }
+
+  auto make_header_model() {
+    auto model = std::make_shared<ArrayListModel<TableHeaderItem::Model>>();
+    auto push = [&] (const QString& name, const QString& short_name) {
+      model->push({name, short_name, TableHeaderItem::Order::UNORDERED,
+        TableFilter::Filter::NONE});
+    };
+    push(QObject::tr("Time"), QObject::tr("Time"));
+    push(QObject::tr("Price"), QObject::tr("Px"));
+    push(QObject::tr("Size"), QObject::tr("Sz"));
+    push(QObject::tr("Market"), QObject::tr("Mkt"));
+    push(QObject::tr("Condition"), QObject::tr("Cond"));
+    push("", "");
+    return model;
+  }
+
+  auto make_header_item_properties() {
+    auto properties = std::vector<HeaderItemProperties>();
+    properties.emplace_back(false, Qt::AlignLeft, scale_width(48));
+    properties.emplace_back(true, Qt::AlignRight, scale_width(50));
+    properties.emplace_back(true, Qt::AlignRight, scale_width(40));
+    properties.emplace_back(true, Qt::AlignLeft, scale_width(38));
+    properties.emplace_back(false, Qt::AlignLeft, scale_width(34));
+    return properties;
+  }
+
+  auto make_time_cell(ptime time) {
+    auto time_text = to_text(time);
+    time_text = time_text.left(time_text.lastIndexOf('.'));
+    return make_label(time_text);
+  }
+
+  template<typename B, typename T =
+    std::decay_t<decltype(*std::declval<B>().get_current())>::Scalar>
+  auto make_decimal_cell(const T& value) {
+    auto cell = new B();
+    cell->get_current()->set(value);
+    cell->set_read_only(true);
+    update_style(*cell, [] (auto& style) {
+      style.get(Any() > is_a<TextBox>()).
+        set(TextAlign(Qt::AlignRight | Qt::AlignVCenter));
+      });
+    return cell;
+  }
+
+  QWidget* table_view_builder(const std::shared_ptr<TableModel>& table, int row,
+      int column) {
+    auto column_id = static_cast<TimeAndSalesTableModel::Column>(column);
+    auto cell = [&] () -> QWidget* {
+      if(column_id == TimeAndSalesTableModel::Column::TIME) {
+        return make_time_cell(table->get<ptime>(row, column));
+      } else if(column_id == TimeAndSalesTableModel::Column::PRICE) {
+        return make_decimal_cell<MoneyBox>(table->get<Money>(row, column));
+      } else if(column_id == TimeAndSalesTableModel::Column::SIZE) {
+        return make_decimal_cell<QuantityBox>(
+          table->get<Quantity>(row, column));
+      } else if(column_id == TimeAndSalesTableModel::Column::MARKET) {
+        return make_label(
+          QString::fromStdString(table->get<std::string>(row, column)));
+      } else if(column_id == TimeAndSalesTableModel::Column::CONDITION) {
+        return make_label(
+          to_text(table->get<TimeAndSale::Condition>(row, column)));
+      }
+      return make_label("");
+    }();
+    update_style(*cell, [] (auto& style) {
+      apply_table_cell_style(style);
```

Comment:
Factor all of this out, don't create lambda expressions just for the sake of creating a lambda expression.